src/main/java/com/microsoft/aad/msal4j/AcquireTokenSilentSupplier.java

```diff
@@ -52,6 +52,22 @@ AuthenticationResult execute() throws Exception {
                     res.refreshOn() < currTimeStampSec && res.expiresOn() >= currTimeStampSec;
 
             if (silentRequest.parameters().forceRefresh() || afterRefreshOn || StringHelper.isBlank(res.accessToken())) {
+
+                //Set telemetry info for reason behind the refresh
```

Comment:
what does it mean?

Reply:
Newer versions of the telemetry schema have a field for collecting data about why a token was refreshed (i.e., because of force refresh, it expired, there was no access token, etc.), and this if/else block sets the telemetry value based on the reason.

I've clarified that comment in the latest commit.